### PR TITLE
DEV I Post Report System 

### PR DESCRIPTION
**Pull request by cemrekut**
Adding report button for post, admins can delete post with this feature and users can report the posts.

### CODE REVIEW

**Review by MariaAsif135 — APPROVED** [review]
fixed

File: app_server/posts/urls.py [hunk 1]
```diff
@@ -29,4 +30,5 @@
     path("tag/<slug:tag_slug>/", post_list, name="post_list_by_tag"),
     path("like/<int:pk>", toggle_like, name="like_post"),
     path("check_link/", check_link, name="check_link"),
+    path("report_post/<int:pk>", report_post, name="report_post"),
```
Comment (ceydaduzgec) [inline app_server/posts/urls.py:33]: We need slash at the end of the url.

File: app_server/posts/views.py [hunk 2]
```diff
@@ -22,15 +22,20 @@ def check_link(request):
 
 @login_required
 def post_list(request, tag_slug=None):
+    print("post_list")
```
Comment (ceydaduzgec) [inline app_server/posts/views.py:25]: We shouldn't print anything.

File: app_server/posts/views.py [hunk 3]
```diff
@@ -22,15 +22,20 @@ def check_link(request):
...
     user = request.user
     spaces = Space.objects.filter(Q(owner=user) | Q(members=user) | Q(granted_members=user) | Q(moderators=user))
     posts = (
-        Post.objects.filter(Q(spaces__in=spaces) | Q(author__in=user.following.all()) | Q(author=user))
+        Post.objects.filter(Q(spaces_in=spaces) | Q(author_in=user.following.all()) | Q(author=user))
```
Comment (ceydaduzgec) [inline app_server/posts/views.py:29]: This is wrong. Please use double underscore.

File: app_server/posts/views.py [hunk 4]
```diff
@@ -190,12 +207,12 @@ def search(request):
         posts_s = Post.objects.filter(
             Q(title__icontains=searched)
             | Q(text__icontains=searched)
-            | Q(tags__name__icontains=searched)
-            | Q(tag_descriptions__description__icontains=searched)
+            | Q(tags_name_icontains=searched)
+            | Q(tag_descriptions_description_icontains=searched)
```
Comment (ceydaduzgec) [inline app_server/posts/views.py:211]: This is wrong. Please revert back.

File: app_server/posts/views.py [hunk 5]
```diff
@@ -190,12 +207,12 @@ def search(request):
...
         ).distinct()
 
         # Search spaces with the given keyword
-        spaces_s = Space.objects.filter(Q(name__icontains=searched) | Q(description__icontains=searched))
+        spaces_s = Space.objects.filter(Q(name_icontains=searched) | Q(description_icontains=searched))
```
Comment (ceydaduzgec) [inline app_server/posts/views.py:215]: Wrong.

File: app_server/spaces/tests/tests.py [hunk 6]
```diff
@@ -1,98 +0,0 @@
-from django.contrib.auth import get_user_model
```
Comment (ceydaduzgec) [inline app_server/spaces/tests/tests.py:1]: Why did you delete this file?

File: app_server/posts/models.py [hunk 7]
```diff
@@ -51,7 +51,7 @@ class TagDescription(models.Model):
     tag = models.ForeignKey("taggit.Tag", on_delete=models.CASCADE, related_name="tag_description")
     description = models.CharField(max_length=200, blank=True, unique=False)
 
-    def __str__(self):
+    def _str_(self):
```
Comment (ceydaduzgec) [inline app_server/posts/models.py:54]: These are also wrong please revert back to doule underscores.

**Review by ceydaduzgec — COMMENTED** [review]
Please ask for review before merging. We already have so many bugs. We shouldn't add new ones to the system.

**Review by ceydaduzgec — COMMENTED** [review]
Please ask for review before merging. We already have so many bugs. We shouldn't add new ones to the system.
